modules/apps/27-interchain-accounts/controller/ibc_middleware.go

```diff
@@ -250,6 +250,19 @@ func (im IBCMiddleware) OnChanUpgradeRestore(ctx sdk.Context, portID, channelID
 	return nil
 }
 
+// OnChanUpgradeTimeout implements the IBCModule interface
+func (im IBCMiddleware) OnChanUpgradeTimeout(
+	ctx sdk.Context,
+	portID, channelID string,
+	counterpartyChannel channeltypes.Channel,
+	prevErrorReceipt channeltypes.ErrorReceipt,
+	proofCounterpartyChannel,
+	proofErrorReceipt []byte,
+	proofHeight ibcexported.Height,
+) error {
+	return im.app.OnChanUpgradeTimeout(ctx, portID, channelID, counterpartyChannel, prevErrorReceipt, proofCounterpartyChannel, proofErrorReceipt, proofHeight)
```

Comment:
any reason why you forward here instead of nil? Just curious.

Reply:
ah, because we are calling it from the controller middleware here which should pass the call through to the app for the app callback. 

basically, this would be called in the usecase where you have an underlying module that you want to use to authenticate ICA actions (the kinda v0 auth module design we had for ICA), and then you wrap that application with the ICA controller middleware. in the v1 design you can pass a `nil` app to the ICA controller middleware, and use the controller `msg_server` directly.